# Daffa Asyqar Ahmad Khalisheka

# 1103200034

# Machine Learning Class

# Telkom University

In [7]:
# Import library sklearn.datasets untuk mengambil dataset Iris
from sklearn import datasets
# Import library sklearn.model_selection untuk membagi dataset menjadi train set dan test set
from sklearn.model_selection import train_test_split

# Load dataset Iris
iris = datasets.load_iris()
# Menyimpan atribut dataset ke dalam variabel 'atribut'
atribut = iris.data
# Menyimpan label dataset ke dalam variabel 'label'
label = iris.target

# Memisahkan dataset menjadi train set dan test set dengan perbandingan 80:20
X_train, X_test, y_train, y_test = train_test_split(
    atribut, label, test_size=0.2)


Kode di atas digunakan untuk mengimpor dataset Iris menggunakan library sklearn.datasets. Dataset Iris berisi atribut dan label dari bunga iris. Kemudian, kode tersebut membagi dataset ini menjadi dua bagian: train set (X_train dan y_train) dan test set (X_test dan y_test) dengan perbandingan 80:20 menggunakan library sklearn.model_selection.train_test_split. Dengan pemisahan ini, Anda dapat menggunakan train set untuk melatih model dan test set untuk menguji kinerja model tersebut.

Hasil akurasi tanpa PCA

In [8]:
# Mengimpor kelas DecisionTreeClassifier dari modul tree di library scikit-learn
from sklearn import tree

# Membuat objek DecisionTreeClassifier
decision_tree = tree.DecisionTreeClassifier()

# Melatih model pertama menggunakan data pelatihan (X_train dan y_train)
model_pertama = decision_tree.fit(X_train, y_train)

# Menghitung akurasi model pertama menggunakan data pengujian (X_test dan y_test)
akurasi_model_pertama = model_pertama.score(X_test, y_test)


Kode di atas mengilustrasikan penggunaan DecisionTreeClassifier dari library scikit-learn. Langkah-langkahnya adalah sebagai berikut:

1. Mengimpor kelas DecisionTreeClassifier dari modul tree di library scikit-learn.
2. Membuat objek DecisionTreeClassifier dengan nama `decision_tree`.
3. Melatih model pertama dengan menggunakan data pelatihan (X_train dan y_train) menggunakan metode `fit()`.
4. Menghitung akurasi model pertama dengan menggunakan data pengujian (X_test dan y_test) menggunakan metode `score()`. Akurasi ini mengukur sejauh mana model pertama mampu memprediksi dengan benar berdasarkan data pengujian yang diberikan.

In [4]:
from sklearn.decomposition import PCA

#membuat objek PCA dengan 4 principal component
pca = PCA(n_components=4)

#mengaplikasikan PCA pada dataset
pca_attributes = pca.fit_transform(X_train)

#melihat variance dari setiap atribut
pca.explained_variance_ratio_

array([0.9233377 , 0.05444316, 0.01704481, 0.00517433])

Kode di atas mengilustrasikan penggunaan PCA (Principal Component Analysis) dalam machine learning dengan library scikit-learn (sklearn) di Python. PCA adalah metode untuk mengurangi dimensi data dengan memproyeksikannya ke komponen utama yang paling penting. Dalam contoh ini:

1. Mengimpor modul PCA dari sklearn.
2. Membuat objek PCA dengan spesifikasi bahwa kita ingin menghasilkan 4 komponen utama dari data.
3. Mengaplikasikan PCA pada data pelatihan (X_train) dengan menggunakan metode `fit_transform()`.
4. Menghitung variance yang dijelaskan oleh setiap komponen utama dengan menggunakan atribut `explained_variance_ratio_`. Hasilnya adalah array yang menunjukkan bahwa komponen utama pertama (PC1) menjelaskan sebagian besar variasi dalam data (sekitar 92.3%), diikuti oleh PC2 (sekitar 5.4%), PC3 (sekitar 1.7%), dan PC4 (sekitar 0.5%). Ini membantu kita memahami kontribusi masing-masing komponen utama terhadap variasi data.

Ambil atribut yang memiliki variance 0.921 dan 0.05

In [5]:
#PCA dengan 2 principal component
pca = PCA(n_components=2)
X_train_pca = pca.fit_transform(X_train)
X_test_pca = pca.fit_transform(X_test)

Kode di atas menggunakan metode Principal Component Analysis (PCA) untuk mengurangi dimensi data menjadi dua komponen utama. Ini dilakukan dengan menggunakan objek PCA dengan n_components=2, yang berarti kita ingin menghasilkan dua komponen utama dari data asli. Data pelatihan (X_train) dan data pengujian (X_test) diubah ke dalam representasi yang lebih sederhana dengan hanya dua komponen utama menggunakan metode fit_transform.

In [6]:
#uji akurasi classifier
model2 = decision_tree.fit(X_train_pca, y_train)
model2.score(X_test_pca, y_test)

0.9333333333333333

Akurasi dari classifier model yang digunakan adalah sekitar 93.33%.

Keterangan: Hanya dengan dua atribut kita telah mendapatkan akurasi yang tinggi. Denan atribut yang lebih sedikit, proses yang dijalankan menjadi singkat.